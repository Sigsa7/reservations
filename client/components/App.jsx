import React from 'react';
import axios from 'axios';

import {
  AppWrapper,
  ContentWrapper,
  GlobalStyles,
  ModuleTitle,
} from '../theme/global';

import ReservationForm from './ReservationForm.jsx';
import ListTimes from './actions/ListTimes.jsx';
import ShowButton from './actions/ShowButton.jsx';
import InfoBox from './InfoBox.jsx';

import timeUtils from '../utils/timeUtils';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      restaurantId: Math.floor(Math.random() * 100),
      restaurant: {
        id: -1,
        openTime: {
          hour: 0,
          minute: 0,
        },
        closeTime: {
          hour: 0,
          minute: 0,
        },
        timeIntervals: 0,
        maxSeating: 0,
        maxPartySize: 0,
      },
      bookingsToday: 0,
      available: [],
    };
  }

  componentDidMount() {
    const { restaurantId } = this.state;

    axios.get(`http://localhost:3002/${restaurantId}/reservations`)
      .then((response) => {
        const { restaurant_information } = response.data;
        let { bookings } = response.data;

        const restaurant = {
          id: restaurant_information._id,
          openTime: restaurant_information.open_time,
          closeTime: restaurant_information.close_time,
          timeIntervals: restaurant_information.time_intervals,
          maxSeating: restaurant_information.max_seating,
          maxPartySize: restaurant_information.max_party_size,
          maxTimeRange: restaurant_information.max_time_range,
        };

        bookings = bookings.map(booking => ({
          bookingTime: booking.booking_time,
          partyQty: booking.party_qty,
        }));

        this.setState({
          restaurant,
          available: timeUtils.getAvailableFromBookings(restaurant, bookings),
        });
      })
      .catch((err) => {
        console.log(err);
      });

    axios.get(`http://localhost:3002/${restaurantId}/reservations/count`)
      .then((response) => {
        this.setState({
          bookingsToday: response.data.bookings_count,
        });
      });
  }

  render() {
    const { restaurant, available, bookingsToday } = this.state;
    return (
      <div>
        <GlobalStyles />
        <AppWrapper>
          <ContentWrapper>
            <div>
              <ModuleTitle>
                <span>Make a reservation</span>
              </ModuleTitle>
            </div>
            <ReservationForm restaurant={restaurant} />
            {(available.length > 0) ? <ListTimes available={available} /> : <ShowButton />}
            <InfoBox bookingsToday={bookingsToday} availableCount={available.length} />
          </ContentWrapper>
        </AppWrapper>
      </div>
    );
  }
}

export default App;
