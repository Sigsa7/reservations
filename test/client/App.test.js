import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../client/components/App';

describe('App component', () => {
  it('renders all three sections', () => {
    const component = renderer.create(<App />);
    const json = component.toJSON();
    expect(json.children[0].children).toHaveLength(3);
    expect(json).toMatchSnapshot();
  });
});
