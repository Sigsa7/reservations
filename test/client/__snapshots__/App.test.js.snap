// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders title, form, and times sections 1`] = `
<div>
  <section
    className="theme__Wrapper-qp42uy-0 igZQuB"
  >
    <h1>
      Make a reservation
    </h1>
    <div>
      {"open_time":{"hour":0,"minute":0},"close_time":{"hour":0,"minute":0},"time_intervals":0,"max_seating":0,"max_party_size":0}
    </div>
    <div>
      [{"booking_time":null,"party_qty":0}]
    </div>
  </section>
</div>
`;
